client/network/src/service/tests/chain_sync.rs

```diff
@@ -225,11 +212,8 @@ async fn on_block_finalized() {
 	// send "set sync fork request" message and poll the network
 	network.network().on_block_finalized(hash, header);
 
-	futures::future::poll_fn(|cx| {
-		let _ = network.network().poll_unpin(cx);
-		Poll::Ready(())
-	})
-	.await;
+	// perform one action on network
+	let _ = network.network().next_action();
```

Comment:
Let me be your rustc `futures are not doing anything without polling them`

Reply:
Done